ci-operator/step-registry/upi/install/heterogeneous/powervs/upi-install-heterogeneous-powervs-commands.sh

```diff
@@ -0,0 +1,156 @@
+    exit 0
+fi
+
+echo "Additional worker vm type is ${ADDITIONAL_WORKER_VM_TYPE}"
+echo "Shared dir is ${SHARED_DIR} and cluster profile dir is ${CLUSTER_PROFILE_DIR}"
+
+SHARED_DIR_FILES=$(ls ${SHARED_DIR})
+CLUSTER_PROFILE_DIR_FILES=$(ls ${CLUSTER_PROFILE_DIR})
+echo "Contents of shared dir are ${SHARED_DIR_FILES}"
+echo "Contents of cluster profile are ${CLUSTER_PROFILE_DIR_FILES}"
+#CLUSTER_NAME=$(<"${SHARED_DIR}/cluster_name")
+#BASE_DOMAIN=$(<"${CLUSTER_PROFILE_DIR}/base_domain")
+
+#echo "Cluster name and Base Domain is ${CLUSTER_NAME} ${BASE_DOMAIN}"
+
+function approve_csrs() {
+  while [[ ! -f '/tmp/scale-out-complete' ]]; do
+    sleep 30
+    echo "approve_csrs() running..."
+    oc get csr -o go-template='{{range .items}}{{if not .status}}{{.metadata.name}}{{"\n"}}{{end}}{{end}}' \
+      | xargs --no-run-if-empty oc adm certificate approve || true
+  done
+}
+
+function get_ready_nodes_count() {
+  oc get nodes \
+    -o jsonpath='{range .items[*]}{.metadata.name}{","}{.status.conditions[?(@.type=="Ready")].status}{"\n"}{end}' | \
+    grep -c -E ",True$"
+}
+
+# wait_for_nodes_readiness loops until the number of ready nodes objects is equal to the desired one
+function wait_for_nodes_readiness()
+{
+  local expected_nodes=${1}
+  local max_retries=${2:-10}
+  local period=${3:-5}
+  for i in $(seq 1 "${max_retries}") max; do
+    if [ "${i}" == "max" ]; then
+      echo "[ERROR] Timeout reached. ${expected_nodes} ready nodes expected, found ${ready_nodes}... Failing."
+      return 1
+    fi
+    sleep "${period}m"
+    ready_nodes=$(get_ready_nodes_count)
+    if [ x"${ready_nodes}" == x"${expected_nodes}" ]; then
+        echo "[INFO] Found ${ready_nodes}/${expected_nodes} ready nodes, continuing..."
+        return 0
+    fi
+    echo "[INFO] - ${expected_nodes} ready nodes expected, found ${ready_nodes}..." \
+      "Waiting ${period}min before retrying (timeout in $(( (max_retries - i) * (period) ))min)..."
+  done
+}
+
+EXPECTED_NODES=$(( $(get_ready_nodes_count) + ADDITIONAL_WORKERS ))
+
+echo "Cluster type is ${CLUSTER_TYPE}"
+
+case "$CLUSTER_TYPE" in
+*ibmcloud*)
+  # Add code for ppc64le
+  if [ "${ADDITIONAL_WORKER_ARCHITECTURE}" == "ppc64le" ]; then
+      echo "Adding additional ppc64le nodes"
+      IBMCLOUD_HOME_FOLDER=/tmp/ibmcloud
+      SERVICE_NAME=power-iaas
+      SERVICE_PLAN_NAME=power-virtual-server-group
+      WORKSPACE_NAME=rdr-mac-$REGION-n1 # TODO: Should this be an env variable or some randomly generated name based on the zone?
+      OPENSHIFT_CLIENT_TARBALL=""
+
+      mkdir -p ${IBMCLOUD_HOME_FOLDER}
+      if [ -z "$(command -v ibmcloud)" ]; then
+        echo "ibmcloud CLI doesn't exist, installing"
+        curl -fsSL https://clis.cloud.ibm.com/install/linux | sh
+      fi
+
+      function ic() {
+        HOME=${IBMCLOUD_HOME_FOLDER} ibmcloud "$@"
+      }
+
+      ic version
+      ic login --apikey @${CLUSTER_PROFILE_DIR}/ibmcloud-api-key -r ${REGION}
```

Comment:
I think we could update the create-var-file.sh

Reply:
This is used to create the workspace in powervs